### PR TITLE
nixops: fix dependencies

### LINKED ISSUE #147034: nixops fails to build.
### Steps To Reproduce
1. ...nix-shell -p nixops
2. ...
3. ...

### Expected behavior
it builds!

### Additional context

error: builder for '/nix/store/48sdc9gb26pb3prykylvy644mmsma0km-python2.7-ruamel-yaml-clib-0.2.4.drv' failed with exit code 1;

https://hydra.nixos.org/build/157985763

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.70, NixOS, 21.05.20211003.7daf355 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210922_bcd73eb`
 - channels(user): `"nixgl"`
 - channels(root): `"nixos-21.11pre332093.7fad01d9d5a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### LINKED ISSUE #148194: python2.7-flake8-4.0.1 failed to build
### Describe the bug

```
builder for '/nix/store/3iwy0v935l5g8q8ggw6sgravncsjlq69-python2.7-flake8-4.0.1.drv' failed with exit code 1

cannot build derivation '/nix/store/6ckahsfycginmmjhzk8yfxf1j0rxphvw-python2.7-html5lib-1.1.drv': 1 dependencies couldn't be built
building '/nix/store/q969hll6hpjwz47i7l3669xafq8dzhd3-python2.7-olefile-0.46.drv'...
cannot build derivation '/nix/store/6n0nclmw1275qwis27yxxdx8zc91p91m-python2.7-sphinx-1.8.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/l8zjlpbk3l6lcjc788gx2p86cjkg5dir-python2.7-arrow-0.17.0.drv': 1 dependencies couldn't be built
building '/nix/store/vh3q859xljmnxmybwkah7pfcmvpc0451-python2.7-pyroma-2.6.1.drv'...
cannot build derivation '/nix/store/9w5ld179l8w0iyff57ws3flszpph1fyy-python2.7-construct-2.10.54.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r18kc25n6ak41zix5yrld688r5lgwa59-python2.7-brotlipy-0.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/sqnfrh6apyrdy2cnkrhan4i16n7cirrn-python2.7-httpbin-0.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/83knpz4yzpbjlg698bcs33yhkjczq84c-python2.7-pytest-httpbin-1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/50k88ikb7k805q7ls9i8pls2mwa8r7hp-python2.7-vcrpy-3.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xn2qqm6hpmbnrdqxx29jgxf3kqvdsra9-python2.7-datadog-0.42.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/zshvfjp2hqrybhsyc0lg6dzm2mdmy4ds-nixops-1.7.drv': 1 dependencies couldn't be built
```

### Steps To Reproduce

[root@G6400:/etc/nixos]# nixos-rebuild boot

### Metadata
[root@G6400:/etc/nixos]# nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.8, NixOS, 21.05.2329.348bc5de8bc (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11.333823.96b4157790f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/147034
#144627 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nixinator — APPROVED** [review]
LGTM!

**Review by mweinelt — APPROVED** [review]

File: pkgs/development/python-modules/construct/2.10.54.nix [hunk 1]
```diff
@@ -14,9 +14,20 @@ buildPythonPackage rec {
...
+    "tests/gallery/test_gallery.py"
+    "tests/test_benchmarks.py"
+    "tests/test_compiler.py"
+  ];
+
+  disabledTests = [
+    "test_timestamp"
+  ] ++ (lib.optionals stdenv.isDarwin [
+    "test_multiprocessing"
+  ]);
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/construct/2.10.54.nix:30]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
    "test_multiprocessing"
  ];
```
nit: Those are optional.

Comment (dotlambda) [reply pkgs/development/python-modules/construct/2.10.54.nix]: True. I have to remember they're only needed for `with`.

**Review by jonringer — APPROVED** [review]

**Comment by roberth** [comment]
@ofborg build nixops

**Comment by mweinelt** [comment]
> @ofborg build nixops

Not sure you are aware of what this is testing.

https://github.com/NixOS/nixpkgs/blob/5cf75a4c665161ca6b3788061750f40ce0ea01aa/nixos/tests/nixops/default.nix#L5-L15

**Comment by roberth** [comment]
> Not sure you are aware of what this is testing.

I was just building. I'm aware because I wrote that test.

**Comment by roberth** [comment]
The darwin builds do still fail.

**Comment by dotlambda** [comment]
> The darwin builds do still fail.

I won't be fixing that cause I can't test on Darwin.

**Comment by roberth** [comment]
Alright, then I'll merge this. Thank you @dotlambda!

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/issues/148194

**Comment by dotlambda** [comment]
@NixOS/nixops-committers Someone needs to backport this to 21.11.

**Comment by roberth** [comment]
Oops, wrong label. I'll change the generated PR base to fix that.

The backport action failed last time for some reason. Let's see if it works now.

**Pull request merged**